client/checkout/woopay/direct-checkout/index.js

```diff
@@ -9,59 +9,116 @@ import { debounce } from 'lodash';
-
-window.addEventListener( 'load', async () => {
-	if (
-		! WooPayDirectCheckout.isWooPayDirectCheckoutEnabled() ||
-		shouldSkipWooPay()
-	) {
+/**
+ * Handle the WooPay direct checkout for the given checkout buttons.
+ *
+ * @param {HTMLElement[]} checkoutButtons An array of checkout button elements.
+ */
+const handleWooPayDirectCheckout = async ( checkoutButtons ) => {
+	if ( ! checkoutButtons ) {
 		return;
 	}
 
-	WooPayDirectCheckout.init();
-
-	isThirdPartyCookieEnabled = await WooPayDirectCheckout.isWooPayThirdPartyCookiesEnabled();
-	const checkoutElements = WooPayDirectCheckout.getCheckoutRedirectElements();
+	const isThirdPartyCookieEnabled = await WooPayDirectCheckout.isWooPayThirdPartyCookiesEnabled();
 	if ( isThirdPartyCookieEnabled ) {
 		if ( await WooPayDirectCheckout.isUserLoggedIn() ) {
 			WooPayDirectCheckout.maybePrefetchEncryptedSessionData();
-			WooPayDirectCheckout.redirectToWooPay( checkoutElements, true );
+			WooPayDirectCheckout.addRedirectToWooPayEventListener(
+				checkoutButtons,
+				true
+			);
 		}
 
 		return;
 	}
 
 	// Pass false to indicate we are not sure if the user is logged in or not.
-	WooPayDirectCheckout.redirectToWooPay( checkoutElements, false );
+	WooPayDirectCheckout.addRedirectToWooPayEventListener(
+		checkoutButtons,
+		false
+	);
+};
+
+/**
+ * Add an event listener to the mini cart checkout button.
+ */
+const addMiniCartEventListener = async () => {
+	const checkoutButton = WooPayDirectCheckout.getMiniCartProceedToCheckoutButton();
+	handleWooPayDirectCheckout( [ checkoutButton ] );
+};
+
+/**
+ * If the mini cart widget is available on the page, observe when the drawer element gets added to the DOM.
+ *
+ * As of today, no window events are triggered when the mini cart is opened or closed,
+ * nor there are attribute changes to the "open" button, so we have to rely on a MutationObserver
+ * attached to the `document.body`, which is where the mini cart drawer element is added.
+ */
+const maybeObserveMiniCart = () => {
+	// Check if the widget is available on the page.
+	if ( ! document.querySelector( '.wc-block-mini-cart' ) ) {
+		return;
+	}
+
+	// Create a MutationObserver to check when the mini cart drawer is added to the DOM.
+	const observer = new MutationObserver( ( mutations ) => {
+		for ( const mutation of mutations ) {
+			if ( mutation?.addedNodes?.length > 0 ) {
+				for ( const node of mutation.addedNodes ) {
+					// Check if the mini cart drawer parent selector was added to the DOM.
+					if (
+						node.nodeType === 1 &&
+						node.matches(
+							'.wc-block-components-drawer__screen-overlay'
+						)
+					) {
+						// Wait until the button is rendered and add the event listener to it.
+						waitForSelector(
+							WooPayDirectCheckout.redirectElements
+								.BLOCKS_MINI_CART_PROCEED_BUTTON,
+							addMiniCartEventListener
+						);
+						return;
+					}
+				}
+			}
+		}
+	} );
+
+	observer.observe( document.body, { childList: true } );
```

Comment:
Alternatively, we could add a click event listener to the mini cart button, but that would fail if the button gets re-rendered, so I figured the most reliable way would be through a `MutationObserver`.

There can be some performance overhead to attach the observer to `document.body` in case the website has too many DOM changes constantly going on for some reason, but we're doing our best here by passing only `childList` as `true`, and not `subtree` or `attributes` types of events.

So basically, we're only iterating nodes that are being added or removed from `document.body` and that are immediate children.